solidity/contracts/token/TokenBridgeCctp.sol

```diff
@@ -137,8 +140,28 @@ contract TokenBridgeCctp is HypERC20Collateral, AbstractCcipReadIsm {
             (bytes, bytes)
         );
 
+        bytes calldata tokenMessage = _hyperlaneMessage.body();
+        require(
+            tokenMessage.length == CCTP_TOKEN_BRIDGE_MESSAGE_LEN,
+            "Invalid message body length"
+        );
+
         bytes29 originalMsg = TypedMemView.ref(cctpMessage, 0);
 
+        bytes32 sourceSender = originalMsg._sender();
+        require(sourceSender == _hyperlaneMessage.sender(), "Invalid sender");
+
+        bytes29 burnMessage = originalMsg._messageBody();
+        require(
+            TokenMessage.amount(tokenMessage) == burnMessage._getAmount(),
```

Comment:
Can we add `TokenMessage` functions to `bytes` with `using TokenMessage for bytes`?

Reply:
this wasnt possible because `recipient` is ambiguous for `Message` and `TokenMessage`